common-money.lic

```diff
@@ -48,18 +50,40 @@ module DRCM
   end
 
   def withdraw_exact_amount?(amount_as_string, hometown)
-    bank_id = get_data('town')[hometown]['deposit']['id']
+    settings = get_settings
```

Comment:
Its a common pattern in the common files to not use get_settings. It would be better to change the signature of the method and pass in the settings you need

Reply:
@rcuhljr or @rpherbig might be able to refresh my memory on the reason for this, its escaping me at the moment.

Reply:
The problem with calling `get_settings` here is that it will only get your settings from `Foo-setup.yaml`. That is, any args/additional YAMLs that the calling script might be using won't be loaded, which means the two could be acting on different settings.

Reply:
It only needs to get the data on initialize unless you change it while on and then ;k common-money and ;common-money fix it. The bankbot settings are in `foo-setup.yaml` and no where else.

Reply:
It will get potentially-out-of-sync settings every time the function runs. We don't know in which YAML any given person's bankbot settings will live.

Reply:
The 'main' settings are in base.yaml. Why would someone put the settings in a different yaml other than the setup one?

Reply:
I'm continually impressed at the creative ways people come up with to organize their YAMLs

Reply:
How should we proceed then? Pass the data into common-money? That would require whatever calls common-money to have the variables added in each script to pass.

Reply:
Yeah, I actually agree here a lot. If they can mess it up, they will. Think that's the best way to look at development. That one one of my toughest lessons, learning how to break my own code by doing stupid things users would do.